ml_garden/implementation/tabular/xgboost/model.py

```diff
@@ -1,79 +1,17 @@
-# %%
 import logging
-import time
-import zipfile
-from os import unlink
 
 import pandas as pd
 import xgboost as xgb
 
+from ml_garden.core.constants import Task
 from ml_garden.core.model import Model
-from ml_garden.utils import compression_utils
 
 logger = logging.getLogger(__file__)
 
 
-# %%
-# Function to save and compress an XGBoost model
-def save_and_compress_xgb_booster(
```

Comment:
Maybe we could have an `XGBoostBase` class. It would have the fit and predict methods which are the same between XGBoostClassifier and XGBoostRegressor and move these two as methods of the class.